src/main/java/org/opentripplanner/routing/carspeed/WeekBaselineCarSpeedProvider.java

```diff
@@ -0,0 +1,74 @@
+/* This program is free software: you can redistribute it and/or
+ modify it under the terms of the GNU Lesser General Public License
+ as published by the Free Software Foundation, either version 3 of
+ the License, or (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>. */
+
+package org.opentripplanner.routing.carspeed;
+
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.TreeMap;
+import java.util.concurrent.TimeUnit;
+
+import org.opentripplanner.routing.edgetype.StreetEdge;
+
+/**
+ * Tentative implementation for a week baseline with variable slots size.
+ *
+ * TODO Handle current/predicted values
+ */
+public class WeekBaselineCarSpeedProvider implements StreetEdgeCarSpeedProvider {
+
+    private final float averageSpeed;
+
+    private final NavigableMap<Integer, Float> carSpeeds;
```

Comment:
I'm concerned that we're using Integers and Floats here; that uses a lot more memory than we need to. We should use a Trove TIntFloatMap instead (with NaN instead of null for no data)